bitsail-connectors/connector-doris/src/main/java/com/bytedance/bitsail/connector/doris/sink/DorisSink.java

```diff
@@ -137,16 +144,38 @@ private void initDorisOptions(BitSailConfiguration writerConfiguration) {
     // Need partition info in batch replace modes.
     if (isHasPartition && this.writeMode.equals(DorisExecutionOptions.WRITE_MODE.BATCH_REPLACE)) {
       //BATCH and REPLACE mode need the partition infos
-      List<Map<String, Object>> partitionList = writerConfiguration.getNecessaryOption(DorisWriterOptions.PARTITIONS, CommonErrorCode.CONFIG_ERROR);
-      builder.partitions(
-          partitionList.stream()
-              .map(partition -> JSON.parseObject(JSON.toJSONString(partition), DorisPartition.class))
-              .collect(Collectors.toList())
-      );
+      DorisPartitionTemplate dorisPartitionTemplate = writerConfiguration.getNecessaryOption(DorisWriterOptions.PARTITIONS, CommonErrorCode.CONFIG_ERROR);
+      List<DorisPartition> dorisPartitions = parseTemplateToDorisPartitions(dorisPartitionTemplate);
```

Comment:
Great work, but I have one question.

- Old partition option support flexible partition range. For example, ["start_range": "2022-01-01", "end_range": "2022-02-01"].
- In this PR, `DorisSink` always generate  partitions of single day.
- So could you add a switch option to decide if using this template?

Reply:
Thanks for your suggestion, now I understand what you mean.
So if I add a parameter representing the day, week, month, quarter and year to the template to represent the step size of the partition, is it okay? I think there are very few scenes  partitioned by a random number of days  like:  4 days or 11 days

Reply:
Actually, I mean we can keep both two method for setting doris partition(s). 
 - The first way is deducing partition as your codes.
 - And the second way is the original method, _i.e., directly setting doris partition.
 
So we don't need a "step size", but an option to decide which way to use.

Reply:
Like this:
```
bool useDeducedPartition = xxx // get from job configuration
if (useDeducedPartition) {
   // use your codes
} else {
   // use the original codes
}
```

Reply:
OK，ths!